### PR TITLE
drivers: serial: uart_sam0: Fix tx interrupt behaviour

### PR DESCRIPTION
**Pull request by ghost**
Interrupt driven uart on sam0 currently returns true when uart_irq_tx_ready is called even if tx interrupts are not yet enabled.
This is because per the datasheet the DRE bit is always set to 1 when the DATA register is empty. This causes drivers such as [modbus_serial.c](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/modbus/modbus_serial.c#L422) to corrupt received data since it checks both uart_irq_rx_ready then uart_irq_tx_ready in that order,.

RX data is read, loaded into the shared buffer, pointer incremented, then TX is called (even though it should not be at this moment) and data is sent out and the pointer is [again incremented](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/modbus/modbus_serial.c#L394) resulting in a uart buffer overrun and RX data corruption.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Comment by dcpleung** [comment]
The compliance check is probably going to fail. So please fix those. It is going to complain about signed-off line, and possibly the commit message summary not being formatted correctly.

File: drivers/serial/uart_sam0.c [hunk 1]
```diff
@@ -775,7 +775,7 @@ static int uart_sam0_irq_rx_ready(const struct device *dev)
 {
 	SercomUsart * const regs = DEV_CFG(dev)->regs;
 
-	return regs->INTFLAG.bit.RXC != 0;
+	return (regs->INTFLAG.bit.RXC != 0) && (regs->INTENSET.bit.RXC != 0);
```
Comment (jfischer-no) [inline drivers/serial/uart_sam0.c:778]: If I remember correctly, that is not right at least for rx_ready, see my comment for CDC ACM, https://github.com/zephyrproject-rtos/zephyr/pull/38155#discussion_r699273701

Comment (rocky134) [reply drivers/serial/uart_sam0.c:778]: Makes sense.  When I get a chance, I'll update to not change this line later today.

Comment (rocky134) [reply drivers/serial/uart_sam0.c:778]: Tested without this change and TX transmission works as expected and RX is not corrupted.
I also verified that when transmitting writes to DATA, it does not trigger a RXC INTFLAG interrupt.

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**